packages/core/src/sso/SamlConnector/utils.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  socialUserInfoGuard,
+  type SetSession,
+} from '@logto/connector-kit';
+import { XMLValidator } from 'fast-xml-parser';
```

Comment:
can we rely on samlify to validate the XML instead of introduce new dependencies?

Reply:
Seems not, introducing another XML validator is given as a code example of `samlify` lib.

Reply:
<img width="905" alt="image" src="https://github.com/logto-io/logto/assets/36393111/d4c9ac80-71c4-48fa-a5f7-935e32f8e752">
How about just pick a validator?

Reply:
It comes to my mind that we did not use this validator since they can not work properly when I was debugging locally. Just keep the current one and revisit later?

Reply:
What problem have you met. `authenio/samlify-xsd-schema-validator` works for me when implementing the SAML IdP. 
Anyway, even with `fast-xml-parser'`, can we use samlify as the SSOT for parsing XML files? Keep  `XMLValidator` only be referenced when initializing a samlify instance?

Reply:
Updated